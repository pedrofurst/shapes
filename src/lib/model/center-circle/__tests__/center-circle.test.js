import CenterCircle from '../index';
import Parallelogram from '../../parallelogram';

const newPoint = (x, y) => ({x, y});

const POINT_A = newPoint(2, 4);
const POINT_B = newPoint(3, 3);
const POINT_C = newPoint(1, 2);
const DEFAULT_POINTS = [POINT_A, POINT_B, POINT_C];

describe('Center circle class', () => {
  it('Should draw the center circle', () => {
    const canvas = document.createElement('canvas');
    canvas.setAttribute('width', '200');
    canvas.setAttribute('height', '200');
    const ctx = canvas.getContext('2d');
    const parallelogram = new Parallelogram(DEFAULT_POINTS);
    const centerCircle = new CenterCircle(parallelogram, ctx);
    centerCircle.draw();
    expect(ctx.__getDrawCalls()).toMatchSnapshot();
  });
});

