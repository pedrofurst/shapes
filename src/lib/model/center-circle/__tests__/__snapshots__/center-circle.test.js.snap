// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center circle class Should draw the center circle 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 1,
            "startAngle": 0,
            "x": 1.5,
            "y": 3,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;
